src/cryptonote_core/sent_transition/sent_transition.cpp

```diff
@@ -811,9 +821,42 @@ void transition(
                 final_allocation_before_distrib,
                 unallocated);
 
+    // When we collect stake metadata (like the individual contributions of a user, we store them
+    // into the database and have a unique clause on the (block, tx, contribution index) which
+    // prevents duplicates from being submitted.
+    //
+    // These are typically extracted from an individual smart contract event witnessed on Arbitrum
+    // and they get inserted into a TX. For HF21 we initially bootstrap the amount of locked tokens
+    // by a user by their participation in the $OXEN->$SESH migration. These locked amounts don't
+    // have a transaction associated with them. The DB is written to directly, so to encode these
+    // stakes we set a synthetic TX index which is guaranteed to not conflict with any other TXs in
+    // the block itself.
+    uint32_t synthetic_tx_index = static_cast<uint32_t>(block_tx_count);
+
     snl_state.service_nodes_infos.clear();
-    for (auto& it : post_transition_sns)
+    for (auto& it : post_transition_sns) {
+        // Record the stake metadata to the `block_add_result`. The SQL DB will process these when
+        // it's update set is triggered
+        for (size_t contrib_index = 0; contrib_index < it.sn_info->contributors.size(); contrib_index++) {
+            auto contrib_it = it.sn_info->contributors[contrib_index];
```

Comment:
Reference?

Also using "it" as a name (or part of a name) for a concrete value rather than an iterator feels a bit off.